automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceSpec.scala

```diff
@@ -51,13 +58,127 @@ class WorkspaceSpec extends FreeSpec with WebBrowserSpec with WorkspaceFixtures
 
           val detailPage = listPage.openWorkspaceDetails(billingProject, workspaceName).awaitLoaded()
           detailPage.deleteWorkspace().awaitLoaded()
-
           listPage.validateLocation()
           listPage.filter(workspaceName)
           listPage.ui.hasWorkspace(billingProject, workspaceName) shouldBe false
         }
       }
+
+      "should be able to share the workspace" in withWebDriver { implicit driver =>
+        withWorkspace(billingProject, "WorkspaceSpec_share") { workspaceName =>
+          val listPage = signIn(Config.Users.harry)
+
+          val detailPage = listPage.openWorkspaceDetails(billingProject, workspaceName).awaitLoaded()
+          detailPage.share(Config.Users.ron.email, "READER")
+          detailPage.signOut()
+          val listPage2 = signIn(Config.Users.ron)
+          val detailPage2 = listPage2.openWorkspaceDetails(billingProject, workspaceName).awaitLoaded()
+          detailPage2.ui.readAccessLevel() shouldBe WorkspaceAccessLevel.Reader
+        }
+      }
+
+      "should be able to set can share permissions for other (non-owner) users" in withWebDriver {implicit driver =>
```

Comment:
there could be a range of similar tests that verify editing the permissions after the permissions are set and saved. This would test that upon reopening and rerendering, permissions work and are consistent.

Currently, this might not provide sufficient value, but it's something to keep in mind.